csi/volumeutils.py

```diff
@@ -804,22 +809,32 @@ def volume_list(voltype=None):
             get_subdir_virtblock_vols(mntdir, volumes, PV_TYPE_SUBVOL)
         if voltype is None or voltype == PV_TYPE_VIRTBLOCK:
             get_subdir_virtblock_vols(mntdir, volumes, PV_TYPE_VIRTBLOCK)
+        if voltype is None or voltype == PV_TYPE_RAWBLOCK:
+            get_subdir_virtblock_vols(mntdir, volumes, PV_TYPE_RAWBLOCK)
 
     return volumes
 
 
 def mount_volume(pvpath, mountpoint, pvtype, fstype=None):
     """Mount a Volume"""
     # Need this after kube 1.20.0
-    makedirs(mountpoint)
+    if pvtype == PV_TYPE_RAWBLOCK:
+        # Only create the parent directory since the Target
+        # is raw device file.
+        makedirs(os.path.dirname(mountpoint))
+        Path(mountpoint).touch(mode=0o660)
```

Comment:
we need to create a file with provided targetDirectory. if this is doing that then we need to bind mount the stagingPath to the targetPath.

Reply:
`NodeStageVolume` is not implemented in Kadalu CSI, File from Gluster Vol is directly bind mounted as `target_path` as file. Let me explore about `stagingPath`